File: Java/src/main/java/questions/BestTimeToBuyAndSellStockII.java
package questions;

/**
 * Created by user3301 on 9/15/2017.
 */
public class BestTimeToBuyAndSellStockII {

    public int maxProfit(int[] prices) {
        int profit = 0;
        if (prices.length == 0 || prices.length == 1) {
            return profit;
        }
        for (int i = 0; i < prices.length - 1; i++) {
            if (prices[i + 1] > prices[i]) {
                profit += prices[i + 1] - prices[i];
            }
        }
        return profit;
    }
}

File: CSharp/leetcodesln/BestTimeToBuyAndSellStockII.cs
namespace leetcodesln
{
    public class BestTimeToBuyAndSellStockII
    {

        // greedy
        public int MaxProfit(int[] prices)
        {
            int maxProfit = 0;

            for (int i = 1; i < prices.Length; i++)
            {
                if (prices[i] - prices[i - 1] > 0) maxProfit += (prices[i] - prices[i - 1]);
            }
            return maxProfit;
        }
    }
}
